src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
+void LiveRangeDumper::setDumperStartAt(const LiveRangeListIterator liveRangeIt)
+{
+    m_hasLiveRangestoDump = true;
+    m_StartingLiveRange   = liveRangeIt;
+}
+
+//------------------------------------------------------------------------
+// getStartForDump: Return an iterator to the first "VariableLiveRange" edited/added
+//  during the current "BasicBlock"
+//
+// Return Value:
+//  A LiveRangeListIterator to the first "VariableLiveRange" in "LiveRangeList" which
+//  was used during last "BasicBlock".
+//
+LiveRangeListIterator LiveRangeDumper::getStartForDump() const
+{
+    return m_StartingLiveRange;
+}
+
+//------------------------------------------------------------------------
+// hasLiveRangesToDump: Retutn wheter at least a "VariableLiveRange" was alive during
+//  the current "BasicBlock"'s code generation
+//
+// Return Value:
+//  A boolean indicating indicating if there is at least a "VariableLiveRange"
+//  that has been used for the variable during last "BasicBlock".
+//
+bool LiveRangeDumper::hasLiveRangesToDump() const
+{
+    return m_hasLiveRangestoDump;
+}
+#endif // DEBUG
+
+//------------------------------------------------------------------------
+//                      VariableLiveDescriptor
+//------------------------------------------------------------------------
+
+VariableLiveDescriptor::VariableLiveDescriptor(CompAllocator allocator)
+{
+    // Initialize an empty list
+    m_VariableLiveRanges = allocator.allocate<LiveRangeList>(1);
+
+    size_t m_VariableLiveRangesSize = sizeof(*m_VariableLiveRanges);
+    memset(m_VariableLiveRanges, 0, m_VariableLiveRangesSize);
+
+    new (m_VariableLiveRanges, jitstd::placement_t()) LiveRangeList(allocator);
+
+#if DEBUG
+    m_VariableLifeBarrier = allocator.allocate<LiveRangeDumper>(1);
+
+    size_t m_VariableLifeBarrierSize = sizeof(*m_VariableLifeBarrier);
+    memset(m_VariableLifeBarrier, 0, m_VariableLifeBarrierSize);
+
+    new (m_VariableLifeBarrier, jitstd::placement_t()) LiveRangeDumper(m_VariableLiveRanges);
+#endif // DEBUG
+}
+
+VariableLiveDescriptor::~VariableLiveDescriptor()
+{
+    noway_assert(m_VariableLiveRanges != nullptr);
+
+    m_VariableLiveRanges->~list();
+
+#ifdef DEBUG
+    m_VariableLifeBarrier->~LiveRangeDumper();
+#endif // DEBUG
+}
+
+//------------------------------------------------------------------------
+// hasVariableLiveRangeOpen: Return true if the variable is still alive,
+//  false in other case.
+//
+bool VariableLiveDescriptor::hasVariableLiveRangeOpen() const
+{
+    noway_assert(m_VariableLiveRanges != nullptr);
+
+    return !m_VariableLiveRanges->empty() && !m_VariableLiveRanges->back().m_EndEmitLocation.Valid();
+}
+
+//------------------------------------------------------------------------
+// getAmountLiveRanges: Return amount of variable homes reported for this
+//  variable.
+//
+// Return Value:
+//  A boolean.
+//
+size_t VariableLiveDescriptor::getAmountLiveRanges() const
```

Comment:
The list itself is exposed via `getLiveRanges` and this has only one use. I'd just remove it.

Reply:
It would imply doing things like:
> varLiveDsc->getLiveRanges()->size()

Do we agree on having that kind of staff?

Reply:
Yes, I don't see any problem with that. A more pedantic alternative would be not to expose the list at all and instead expose size/begin/end. I don't think that's worthwile. More generally, I think there's too much ceremony for a data structure that's just an array of linked lists and a bit of logic on top of it. Simple things should be simple.